packages/compass-web/scripts/ws-proxy.js

```diff
@@ -12,6 +12,32 @@ function serializeError(err) {
   }
 }
 
+const certsMap = new Map();
+
+const WEB_PROXY_PORT = process.env.COMPASS_WEB_HTTP_PROXY_PORT
+  ? Number(process.env.COMPASS_WEB_HTTP_PROXY_PORT)
+  : 7777;
+
+async function resolveX509Cert({ projectId, clusterName }) {
+  if (certsMap.has(projectId)) {
+    return certsMap.get(projectId);
+  }
+  try {
+    const cert = await fetch(
+      `http://localhost:${WEB_PROXY_PORT}/x509?projectId=${projectId}&clusterName=${clusterName}`
```

Comment:
Yep, it's not, but also you're totally right, I'll change this to use url constructor just in case somebody might look at this code as an example for something else that might land in production

Reply:
Ok, as you prefer, you can also just resolve this then as far as I am concerned